test/plugin/base_test.rb

```diff
@@ -544,6 +544,33 @@ def test_struct_payload_log
     end
   end
 
+  def test_struct_payload_malformat_log
+    setup_gce_metadata_stubs
+    message = 'test message'
+    setup_logging_stubs do
+      d = create_driver
+      d.emit(
+        'int_as_key' => { 1 => message },
```

Comment:
Key names are changed and a test case for symbol is added. Seems `{ symbol: message }` is preferred in Ruby 1.9 though. (got Rubocop errors for `{ :symbol => message }`